datasets/iwslt2017.py

```diff
@@ -0,0 +1,227 @@
+Address = {Trento, Italy},
+Author = {Mauro Cettolo and Christian Girardi and Marcello Federico},
+Booktitle = {Proceedings of the 16$^{th}$ Conference of the European Association for Machine Translation (EAMT)},
+Date = {28-30},
+Month = {May},
+Pages = {261--268},
+Title = {WIT$^3$: Web Inventory of Transcribed and Translated Talks},
+Year = {2012}}
+"""
+
+_DESCRIPTION = """\
+The IWSLT 2017 Evaluation Campaign includes a multilingual TED Talks MT task. The languages involved are five:
+
+  German, English, Italian, Dutch, Romanian.
+
+For each language pair, training and development sets are available through the entry of the table below: by clicking, an archive will be downloaded which contains the sets and a README file. Numbers in the table refer to millions of units (untokenized words) of the target side of all parallel training sets.
+"""
+
+MULTI_URL = "https://wit3.fbk.eu/archive/2017-01-trnmted//texts/DeEnItNlRo/DeEnItNlRo/DeEnItNlRo-DeEnItNlRo.tgz"
+
+
+class IWSLT2017Config(nlp.BuilderConfig):
+    """ BuilderConfig for NewDataset"""
+
+    def __init__(self, pair, is_multilingual, **kwargs):
+        """
+
+        Args:
+            pair: the language pair to consider
+            **kwargs: keyword arguments forwarded to super.
+        """
+        self.pair = pair
+        self.is_multilingual = is_multilingual
+        super(IWSLT2017Config, self).__init__(**kwargs)
+
+
+# XXX: Artificially removed DE from here, as it also exists within bilingual data
+MULTI_LANGUAGES = ["en", "it", "nl", "ro"]
+BI_LANGUAGES = ["ar", "de", "en", "fr", "ja", "ko", "zh"]
+MULTI_PAIRS = [
+    f"{source}-{target}"
+    for source in MULTI_LANGUAGES
+    for target in MULTI_LANGUAGES
+    if source != target
+]
+BI_PAIRS = [
+    f"{source}-{target}"
+    for source in BI_LANGUAGES
+    for target in BI_LANGUAGES
+    if source != target and (source == "en" or target == "en")
+]
+
+PAIRS = MULTI_PAIRS + BI_PAIRS
+
+
+class IWSLT217(nlp.GeneratorBasedBuilder):
+    """The IWSLT 2017 Evaluation Campaign includes a multilingual TED Talks MT task."""
+
+    VERSION = nlp.Version("1.0.0")
+
+    # This is an example of a dataset with multiple configurations.
+    # If you don't want/need to define several sub-sets in your dataset,
+    # just remove the BUILDER_CONFIG_CLASS and the BUILDER_CONFIGS attributes.
+    BUILDER_CONFIG_CLASS = IWSLT2017Config
+    BUILDER_CONFIGS = [
+        IWSLT2017Config(
+            name="iwslt2017-" + pair,
+            description="A small dataset",
+            pair=pair,
+            is_multilingual=pair in MULTI_PAIRS,
+        )
+        for pair in PAIRS
+    ]
+
+    def _info(self):
+        return nlp.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # nlp.features.FeatureConnectors
+            features=nlp.Features(
+                {"source": nlp.Value("string"), "target": nlp.Value("string"),}
+            ),
```

Comment:
I would use the `Translation` feature type.

You can see an example of usage in the [FLoRes dataset script](https://github.com/huggingface/nlp/blob/master/datasets/flores/flores.py#L93)